### PR TITLE
Create flatcar linux clusters

### PR DESCRIPTION
**Pull request by sowmyav27**
Cluster.yml should be - https://github.com/rancher/rke/issues/1744#issuecomment-597362929

### CODE REVIEW

File: tests/validation/tests/v3_api/test_rke_cluster_provisioning.py [hunk 1]
```diff
@@ -1150,6 +1283,10 @@ def create_custom_host_from_nodes(nodes, node_roles,
         else:
             config = rke_config_windows
 
+    elif NETWORK_PLUGIN == "calico":
+        config = rke_config_flatcar_calico
+    elif NETWORK_PLUGIN == "canal":
+        config = rke_config_flatcar_canal
```
Comment (sangeethah) [inline tests/validation/tests/v3_api/test_rke_cluster_provisioning.py:1289]: Should be done only in the case of flatcar os

File: tests/validation/tests/v3_api/test_rke_cluster_provisioning.py [hunk 2]
```diff
@@ -51,6 +50,132 @@
...
+        "maxUnavailableWorker": "10%"
+    },
+    "nodeDrainInput":{
+        "deleteLocalData": False,
+        "force": False,
+        "gracePeriod": -1,
+        "ignoreDaemonSets": True,
+        "timeout": 120
+    }}
+
```
Comment (sangeethah) [inline tests/validation/tests/v3_api/test_rke_cluster_provisioning.py:178]: flex plugin volume directory can be added as a variable and used for 1 rke_flat_car config.

Comment (sowmyav27) [reply tests/validation/tests/v3_api/test_rke_cluster_provisioning.py:53]: I have added a separate test for test_create_flatcar_linux()

File: tests/validation/tests/v3_api/test_rke_cluster_provisioning.py (hunk 1 above)
Comment (sowmyav27) [reply tests/validation/tests/v3_api/test_rke_cluster_provisioning.py]: Changed this logic

**Review by sangeethah — DISMISSED** [review]
LGTM

**Review by bmdepesa — DISMISSED** [review]
LGTM

**Pull request closed without merging**